### PR TITLE
Add django support, mainly sentry

### PR DESCRIPTION
**Pull request by bloodearnest**
Add support and docs for django. 
Includes a test django project in tests/django_app, which is mainly boilerplate

Also, moved flask support to use custom cls specification rather than the messy get_client(). It now works the same way django did. 

Also, enabled all signals in all celery contexts too, as I realise if a task is .apply()-ed, it runs (with signals) in the wsgi thread.

### CODE REVIEW

File: talisker/sentry.py [hunk 1]
```diff
@@ -58,6 +59,11 @@ def register_client_update(update_func):
     return update_func
 
 
+def update_clients(client):
```
Comment (ricardokirkner) [inline talisker/sentry.py]: should this not better be called update_client (singular)?

File: talisker/django.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import logging
+from raven.contrib.django.client import DjangoClient
+
+import talisker.sentry
+
+
+# raven's django support does some very odd things.
+# There is a module global, that is a proxy to the output of
+# raven.contrib.django.models.get_client()
+# But the only way thing you can customise about the client is it's base class.
```
Comment (bloodearnest) [inline talisker/django.py]: yup, s/thing//

File: talisker/django.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+class SentryClient(DjangoClient):
+    def __init__(self, *args, **kwargs):
+        # SQL hook sends raw SQL to the server. Not cool, bro.
+        # Force disable it for now, until we can add a processors to sanitize
+        kwargs['install_sql_hook'] = False
+        from_env = talisker.sentry.ensure_talisker_config(kwargs)
+        logging.getLogger(__name__).info(
+            'updating raven config from django app')
+        super().__init__(*args, **kwargs)
```
Comment (bloodearnest) [inline talisker/django.py]: It does work in python 2, via the future library, which provides a version for py2 that works.

File: talisker/sentry.py (hunk 1 above)
Comment (bloodearnest) [inline talisker/sentry.py]: Well, you are updating all the references to the client, so I guess update_client_references?

File: talisker/django.py (hunk 3 above)
Comment (bloodearnest) [inline talisker/django.py]: FYI, if the travis tests are passing, then they are passing under python 2.7, 3.4, 3.5

**Pull request merged**